pages/base.py

```diff
@@ -47,57 +47,44 @@ def __getattr__(self, what):
 class BasePage(BaseElement):
     url = None
 
-    def open(self):
+    def goto(self):
         self.driver.get(self.url)
 
+    def verify_page(self):
+        """
+        Use an element from the locator_dictionary to check if we're on expected page
+        """
+        try:
+            self.__getattr__(list(self.locator_dictionary.keys())[0])
+            return True
+        except (TimeoutException,StaleElementReferenceException):
+            return False
+
+    def reload(self):
+        self.driver.refresh()
+
 class OSFBasePage(BasePage):
     url = settings.OSF_HOME
 
     def __init__(self, driver):
         super(OSFBasePage, self).__init__(driver)
         self.navbar = self.Navbar(driver)
 
-    locator_dictionary = {
-        'sign_in_button': (By.LINK_TEXT, 'Sign In')
-    }
-
     def is_logged_in(self):
-        return self.navbar.logged_in()
+        return self.navbar.is_logged_in()
 
 
     class Navbar(BaseElement):
 
         locator_dictionary = {
             'sign_in_button': (By.LINK_TEXT, 'Sign In'),
-            'username_input': (By.ID, 'username'),
-            'password_input': (By.ID, 'password'),
-            'submit_button': (By.NAME, 'submit'),
-            'local_submit_button': (By.ID, 'submit'),
-            'remember_me_checkbox': (By.ID, 'rememberMe'),
             'user_dropdown': (By.CSS_SELECTOR, '#secondary-navigation > ul > li:nth-child(5) > button'),
             'logout_link': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(4) > a')
         }
 
-        def logged_in(self):
+        def is_logged_in(self):
             try:
                 if self.sign_in_button:
                     return False
             except:
```

Comment:
One other thing, in our `__getattr__` method, we only handle two exceptions `StaleElementReferenceException, TimeoutException`, and in both cases we raise a 'ValueError'. Given many other exceptions that may occur (see [here](https://seleniumhq.github.io/selenium/docs/api/py/common/selenium.common.exceptions.html)) are unhandled, we shouldn't just assume any/all exception means the element is not there.